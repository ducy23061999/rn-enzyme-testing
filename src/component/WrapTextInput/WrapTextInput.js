import React, {useEffect, useState} from 'react'
import {
    View,
    Text,
    TextInput,
} from 'react-native'
import styles from './WrapTextInputStyle'
import Utils from '../../Utils'

export default function WrapTextInput({placeholder}) {

    const [isValid, setValid] = useState(false);

    const actions = {
        onTextChange: (text) => {
            const valid = Utils.isValidEmail(text);
            setValid(valid);
        }
    }

    return (
        <View style = {styles.container}>
            <TextInput 
                style = {styles.input}
                placeholder = {placeholder}
                onChangeText = {actions.onTextChange}
            />
            <ValidText isValid = {isValid}/>
        </View>
    )
}

export function ValidText({isValid}) {
    return isValid ? 
        <Text 
            style = {[
                styles.text, 
                styles.valid
            ]}
        >Your text is valid</Text>
        :
        <Text style = {[
            styles.text, 
            styles.inValid
        ]}>Your email you enter wrong</Text>
}
