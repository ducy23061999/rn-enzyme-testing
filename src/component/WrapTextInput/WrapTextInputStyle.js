import {StyleSheet} from 'react-native'
import layout from '../../layout'

export default StyleSheet.create({
    container: {
        ...layout.columnSketchCenter
    },
    input: {
        width: '100%',
        borderColor: '#b3b5b4',
        borderWidth: 2,
        borderRadius: 5
    },
    text: {
        alignSelf: 'flex-start',
        marginTop: 10,
        fontSize: 17
    },
    valid: {
        color: 'green'
    },
    inValid: {
        color: 'red'
    }
})