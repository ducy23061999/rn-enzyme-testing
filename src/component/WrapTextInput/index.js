import WrapTextInput from './WrapTextInput'

export default WrapTextInput;