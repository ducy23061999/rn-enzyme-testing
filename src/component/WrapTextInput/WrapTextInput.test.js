import WrapTextInput, {ValidText} from './WrapTextInput'
import {shallow, mount} from 'enzyme'
import React from 'react'
import {Text, TextInput} from 'react-native'
import 'jsdom-global/register';


describe("WrapTextInput", () => {
    describe("Rendering", () => {
        it("should be render default with invalid email", () => {
            const component = shallow(<WrapTextInput placeholder = "Enter email"/>);
            const InputComponent = component.find(TextInput).first();
            const placeHolder = InputComponent.props().placeholder;
            

            const TextComponent = component.find(ValidText).first();
            const status = TextComponent.props().isValid;
            const numberOfComponent = component.children().length;

            expect(numberOfComponent).toBe(2);
            expect(status).toBeFalsy();
            expect(placeHolder).toEqual("Enter email");
        })
        it("Mount rendering", () => {
            const component = mount(<WrapTextInput placeholder = "Enter email"/>);

            expect(component).toMatchSnapshot();
        })
    })
    describe("Interaction", () => {
        it("should in show valid email", () => {
            const component = shallow(<WrapTextInput placeholder = "Enter email"/>);
            const InputComponent = component.find(TextInput).first();
            InputComponent.simulate('changeText', "ducy20@gmail.com");

            const Text = component.find(ValidText).first();

            const status = Text.props().isValid;

            expect(status).toBeTruthy();

        })
        it("should in show invalid email", () => {
            const component = shallow(<WrapTextInput placeholder = "Enter email"/>);
            const InputComponent = component.find(TextInput).first();
            InputComponent.simulate('changeText', "ducy20");

            const Text = component.find(ValidText).first();
            const status = Text.props().isValid;

            expect(status).toBeFalsy();

        })
    })
})