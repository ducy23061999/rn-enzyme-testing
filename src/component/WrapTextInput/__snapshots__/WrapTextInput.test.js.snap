// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapTextInput Rendering Mount rendering 1`] = `
<WrapTextInput
  placeholder="Enter email"
>
  <View
    style={
      Object {
        "alignContent": "stretch",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "stretch",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Component
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter email"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#b3b5b4",
            "borderRadius": 5,
            "borderWidth": 2,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Enter email"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#b3b5b4",
              "borderRadius": 5,
              "borderWidth": 2,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </Component>
      <ValidText
        isValid={false}
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 17,
                "marginTop": 10,
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "fontSize": 17,
                  "marginTop": 10,
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            Your email you enter wrong
          </Text>
        </Text>
      </ValidText>
    </View>
  </View>
</WrapTextInput>
`;
